### DISCUSSION TITLE
Change order steps into Browser Flow

### DISCUSSION THREAD

**Discussion by adrien-dubois** [keycloak/keycloak#32591] [Q&A]
I just created a new flow binded on Browser Flow, to authenticate my users with a X509 certificate injected into a yubikey.
That works fine, but when users are on the login page, Certificate is asked first, and then, the credentials form displays.
I just want to change that order, so, users have to fill the form, and when they validate, browser ask for certificate.
I tried to change tue order on the flow configuration, add a sub flow to trigger the `Username Password Form` first, and I tried to add a condition too, but browser always ask for certificate first. I don't know how to make it right.
There is a screen of my flow's configuration :
![image](https://github.com/user-attachments/assets/3e42a157-9acc-4d88-ab5b-43f76cf2f714)

If someone have a clue 😅 Thanks

Adrien

**Comment by darius-m** [comment]
I'm not entirely sure what the X509/Validate username step does, but from what you are describing, it sounds like mutual TLS authentication, where the client has to provide a valid certificate to connect to the server during the TLS handshake (otherwise, the connection is dropped during the TLS handshake). If that is the case, it does make sense that the certificate is asked first - and, as a bonus, it should also improve security, as attackers that do not have a valid certificate will not be able to brute force your login form.

However, for a more normal login flow, since you already have yubikeys for authentication, you can also use a WebAuthn step, which will use require the yubikey to sign the message and authenticate.

**Reply by darius-m** [reply]
Here's how it's described in the documentation: https://www.keycloak.org/docs/latest/server_admin/index.html#_x509
